Packages/Tracking/Core/Runtime/Scripts/HandPoses/HandPoseDetector.cs

```diff
@@ -0,0 +1,373 @@
+        [Serializable]
+        public struct FingerDirection
+        {
+            public bool enabled;
+            public TypeOfDirectionCheck typeOfDirectionCheck;
+            public bool isPalmDirection;
+            public Transform poseTarget;
+            public AxisToFace axisToFace;
+            public Finger.FingerType fingerTypeForPoint;
+            public Bone.BoneType boneForPoint;
+            public float rotationThreshold;
+        }
+        #endregion
+
+        public void CreateDefaultFingerDirection()
+        {
+            FingerDirection fingerDirection= new FingerDirection();
+            fingerDirection.typeOfDirectionCheck = TypeOfDirectionCheck.CAMERALOCAL;
+            fingerDirection.isPalmDirection = false;
+            fingerDirection.poseTarget = null;
+            fingerDirection.axisToFace = AxisToFace.Forward;
+            fingerDirection.fingerTypeForPoint = Finger.FingerType.TYPE_INDEX;
+            fingerDirection.boneForPoint = Bone.BoneType.TYPE_DISTAL;
+            fingerDirection.rotationThreshold = 15;
+            fingerDirection.enabled = true;
+
+            BoneDirectionTargets.Add(fingerDirection);
+        }
+        public void RemoveDefaultFingerDirection(int index)
+        {
+            BoneDirectionTargets.RemoveAt(index);
+        }
+
+
+
+        public static event Action<HandPoseScriptableObject> PoseHasBeenDetected;
+        public static event Action PoseHasNotBeenDetected;
+
+        private void Start()
+        {
+            //PoseHasBeenDetected += PoseDetected;
+            //PoseHasNotBeenDetected += PoseNotDetected;
+            if(_leapProvider == null)
+            {
+                _leapProvider = FindObjectOfType<LeapProvider>();
+            }
+        }
+        //private void PoseDetected(HandPoseScriptableObject poseScriptableObject) { }
+        //private void PoseNotDetected() { }
+
+
+        // Update is called once per frame
+        private void Update()
+        {
+            bool anyHandMatched = CompareAllHandsAndPoses();
+            if (anyHandMatched && !_poseAlreadyDetected)
+            {
+                _poseAlreadyDetected = true;
+                PoseHasBeenDetected(detectedPose);
+                Debug.Log("pose Detected");
+            }
+            else if (!anyHandMatched && _poseAlreadyDetected)
+            {
+                _poseAlreadyDetected = false;
+                PoseHasNotBeenDetected();
+                Debug.Log("pose Un Detected");
+            }
+        }
+
+
+
+
+        private bool CompareAllHandsAndPoses()
+        {
+            // If the user hasnt specified the hands to detect, check all Hand Model Bases.
+            // This will only do this once unless manually cleared.
+            detectedPose = null;
+
+            foreach (var activePlayerHand in _leapProvider.CurrentFrame.Hands)
+            {
+                if ((!CheckBothHands && activePlayerHand.GetChirality() == ChiralityToCheck) || CheckBothHands)
```

Comment:
this could be:

`if (CheckBothHands || (activePlayerHand.GetChirality() == ChiralityToCheck))`